### ISSUE TITLE
Testing connected redux components with async actions

### ISSUE THREAD

**Issue by dominictwlee** [kentcdodds/ama#666]
First off, really enjoy using react-testing-library, so thanks for creating such an awesome library!

Following your recommendations on https://www.youtube.com/watch?v=h7ukDItVN_o, I have opted to write integration tests for redux components.

I have a situation where I need to test that the DOM has been updated with the correct values after an async call. 

I use redux middleware to handle my async calls as shown below:
```
const getHeartRatesRequest = (cid: string, startTS: string, endTS: string) => ({
  type: GET_HEARTRATES_REQUEST,
  meta: {
    apiCall: () => consumersApi.getHeartrates(cid, startTS, endTS).then(({ Items }) => Items),
  },
});
```

It's just a standard actionCreator but with a meta tag.

Basically, all the middleware does is take whatever promise is in meta.apiCall and dispatches success/failure actions according to resolve/reject.

What would be a good way for me to handle this test case? I'll need to wait for the middleware to handle the dispatch of actions before checking that my component's DOM has been updated correctly in relation to changes to store state.

Thanks for your time!
